otter/integration/tests/test_convergence.py

```diff
@@ -150,6 +150,55 @@ def wrapper(*args, **kwargs):
         return decorated
 
 
+def tag(*tags):
+    """
+    Decorator that adds tags to a function by setting the property "tags".
+
+    This should be added upstream to Twisted trial.
+    """
+    def decorate(function):
+        function.tags = tags
+        return function
+    return decorate
+
+
+def skip_me(reason):
+    """
+    Decorator that skips a test method or test class by setting the property
+    "skip".  This decorator is not named "skip", because setting "skip" on a
+    module skips the whole tes module.
+
+    This should be added upstream to Twisted trial.
+    """
+    def decorate(function):
+        function.skip = reason
+        return function
+    return decorate
+
+
+def copy_test_methods(from_class, to_class, filter_and_change=None):
+    """
+    Copy test methods (methods that start with `test_*`) from ``from_class`` to
+    ``to_class``.  If a decorator is provided, the test method on the
+    ``to_class`` will first be decorated before being set.
+
+    :param class from_class: The test case to copy from
+    :param class to_class: The test case to copy to
+    :param callable filter_and_change: A function that takes a test name
+        and test method, and returns a tuple of `(name, method)`
+        if the test method should be copied. None else.  This allows the
+        method name to change, the method to be decorated and/or skipped.
+    """
+    for name, attr in from_class.__dict__.items():
```

Comment:
Non-blocking nitpick: Using `dir` function may be better since dunder methods are supposed to be private?